### PR TITLE
Integration of new API

### LINKED ISSUE #175: [BUG] Crash when Fav. Song
A shit, it looks like you did something that the osu!player did not like. If you think this is a bug, please report this crashlog to Founntain by using the buttons below! (Also please don't forget to state how you achieved the crash e. g. what did you do, click etc.)

UTC Time: Wednesday, January 4, 2023 7:29 PM

Error stacktrace below:
Sequence contains no matching element
   at System.Linq.ThrowHelper.ThrowNoMatchException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
   at OsuPlayer.Views.PlaylistViewModel.<.ctor>b__23_3(Object _, PropertyChangedEventArgs _) in C:\Repositories\osuplayer\OsuPlayer\Views\PlaylistViewModel.cs:line 102
   at OsuPlayer.Modules.Audio.Player.TriggerPlaylistChanged(PropertyChangedEventArgs e) in C:\Repositories\osuplayer\OsuPlayer\Modules\Audio\Player.cs:line 190
   at OsuPlayer.Views.PlayerControlView.Favorite_OnClick(Object sender, RoutedEventArgs e) in C:\Repositories\osuplayer\OsuPlayer\Views\PlayerControlView.axaml.cs:line 108
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at Avalonia.Threading.JobRunner.RunJobs(Nullable`1 priority) in /_/src/Avalonia.Base/Threading/JobRunner.cs:line 37
   at Avalonia.Win32.Win32Platform.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam) in /_/src/Windows/Avalonia.Win32/Win32Platform.cs:line 283
   at Avalonia.Win32.Interop.UnmanagedMethods.DispatchMessage(MSG& lpmsg)
   at Avalonia.Win32.Win32Platform.RunLoop(CancellationToken cancellationToken) in /_/src/Windows/Avalonia.Win32/Win32Platform.cs:line 217
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken) in /_/src/Avalonia.Base/Threading/Dispatcher.cs:line 65
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 120
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 209
   at OsuPlayer.Program.Main(String[] args) in C:\Repositories\osuplayer\OsuPlayer\Program.cs:line 30

have fun bitch

### LINKED ISSUE #176: [BUG] Player crashes when trying to play new song with empty playlist
A shit, it looks like you did something that the osu!player did not like. If you think this is a bug, please report this crashlog to Founntain by using the buttons below! (Also please don't forget to state how you achieved the crash e. g. what did you do, click etc.)

UTC Time: Wednesday, January 4, 2023 7:34 PM

Error stacktrace below:
Attempted to divide by zero.
   at OsuPlayer.Modules.Audio.Player.GetNextSongToPlay(IList`1 songSource, Int32 currentIndex, PlayDirection playDirection) in C:\Repositories\osuplayer\OsuPlayer\Modules\Audio\Player.cs:line 359
   at OsuPlayer.Modules.Audio.Player.NextSong(PlayDirection playDirection) in C:\Repositories\osuplayer\OsuPlayer\Modules\Audio\Player.cs:line 277
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at Avalonia.Threading.JobRunner.RunJobs(Nullable`1 priority) in /_/src/Avalonia.Base/Threading/JobRunner.cs:line 37
   at Avalonia.Win32.Win32Platform.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam) in /_/src/Windows/Avalonia.Win32/Win32Platform.cs:line 283
   at Avalonia.Win32.Interop.UnmanagedMethods.DispatchMessage(MSG& lpmsg)
   at Avalonia.Win32.Win32Platform.RunLoop(CancellationToken cancellationToken) in /_/src/Windows/Avalonia.Win32/Win32Platform.cs:line 217
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken) in /_/src/Avalonia.Base/Threading/Dispatcher.cs:line 65
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 120
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 209
   at OsuPlayer.Program.Main(String[] args) in C:\Repositories\osuplayer\OsuPlayer\Program.cs:line 30

### PR DESCRIPTION
**Pull request by Founntain**
Implementation and Integration of the new API and a ton of fixes and changes

fixes #175 
fixes #176

### CODE REVIEW

File: OsuPlayer.Network/API/Service/AbstractApiBase.cs [hunk 1]
```diff
@@ -34,6 +35,10 @@ protected void ParseWebException(Exception ex)
     /// </summary>
     protected void CancelCancellationToken()
     {
+        // TODO: Currently does nothing, because it breaks the functionality a bit. 
+        // TODO: This needs to be re-thinked a bit!
+        return;
+        
         CancellationTokenSource.Cancel();
         CancellationTokenSource = new ();
```
Comment (Cesan) [inline OsuPlayer.Network/API/Service/AbstractApiBase.cs:43]: Commenting these lines out would be better as this code is currently not reachable. The TODO is explaining accordingly.

**Review by Cesan — CHANGES_REQUESTED** [review]

**Comment by Cesan** [comment]
Btw, wont #108 be implemented with this aswell?

**Comment by Founntain** [comment]
Yes you are right. Added it to the linked issues!

**Comment by Founntain** [comment]
Also not forget to mention #70 can actually be started when this PR is done and published as an release

**Review by Cesan — APPROVED** [review]

**Comment by Founntain** [comment]
@Cesan Glad that you looked over the #176 because I wasn't sure about my "fix" because is was gone. Greatful that you checked over it mate!

**Pull request merged**